cdk/v2/webhook/procWorkflow.yaml

```diff
@@ -0,0 +1,73 @@
+bindings:
+  - path: ../../../../v0/util
+  - path: ../../../../v0/destinations/webhook/util
+steps:
+  - name: validateInput
+    template: |
+      (
+        $assert(destination.Config.webhookUrl or false, "Invalid URL in destination config");
+        $assert($exists(message.type), "message Type is not present. Aborting message.");
+      )
+  - name: buildHeaderBlock
+    template: |
+      (
+        $defaultHeaders :=  {"content-type": "application/json"};
+        $finalHeaders := $merge([$defaultHeaders, $getHashFromArray(destination.Config.headers)]);
+        $messageHeader := $type(message.headers) = "object" ? message.headers : {};
+        $finalHeaders := $merge([$finalHeaders, $sift($messageHeader, function($v){(
+          $type($v) = "string"
+          )})]);
+        $setContext("finalHeaders", $finalHeaders);
+      )
+  - name: deduceEndPoint
+    template: |
+      (
+        $integrationsObjects := $getIntegrationsObj(message, "webhook");
+        $defaultEndpoint := destination.Config.webhookUrl;
+        $finaleEndpoint := [ message.fullPath , $integrationsObjects.fullPath, $defaultEndpoint ][0];
+        $messageAppendPath := [ message.appendPath, $integrationsObjects.appendPath ][0];
+        $finaleEndpoint := $finaleEndpoint & $messageAppendPath;
+        $setContext("finaleEndpoint", $finaleEndpoint)
+      )
+  - name: processForGETANDDELETEMethod
+    condition: destination.Config.webhookMethod = $defaultGetRequestConfig.requestMethod or destination.Config.webhookMethod = $defaultDeleteRequestConfig.requestMethod
+    template: |
+      (
+       $response := { 
+        "method" : $defaultGetRequestConfig.requestMethod or $defaultDeleteRequestConfig.requestMethod,
+        "params" : $getPropertyParams(message)
+      };
+      $setContext("payload", $response);
+      )
+     
+  - name: processPOSTPUTPATCHMethod
+    condition: destination.Config.webhookMethod = $defaultPostRequestConfig.requestMethod or destination.Config.webhookMethod = $defaultPutRequestConfig.requestMethod or destination.Config.webhookMethod = $defaultPatchRequestConfig.requestMethod
+    template: |
+      (
+      $userId:= {"userId": message.anonymousId};
+      $response := {
+        "method" : destination.Config.webhookMethod,
+        "body": $merge([$userId,message ~> |$|{}, ['fullPath', 'appendPath']|]),
```

Comment:
saw this https://try.jsonata.org/bL_gYKVOi and implemented. Not quite readable though. @koladilip can you suggest an easier solution?

Reply:
https://try.jsonata.org/a1RhPHPtV

Reply:
> saw this https://try.jsonata.org/bL_gYKVOi and implemented. Not quite readable though. @koladilip can you suggest an easier solution?

I didn't get quite what is the actual goal.

Reply:
@koladilip this is not working. It is generating output like :
```
{
  "context": {
    "ip": "1.1.1.1"
  },
  "request_ip": "2.2.2.2"
}
```
what I need is:
```{
  "context": {
    "ip": "2.2.2.2"
  },
  "request_ip": "2.2.2.2"
}```